src/Photons.ts

```diff
@@ -138,63 +133,66 @@ export default class Photons {
 
   /**
    * Create a propagator, given this object dimX and dimY.
+   * Propagator "moves" photons.
    * @param yDirMeansDown For true, direction 'v' increments dimY.
+   * @todo is yDirMeansDown really necessary? It is only present in this file.
    *
    * @return An operator, with dimensions [dimX, dimY, {@link Dimension.direction()}].
    */
   createPhotonPropagator(yDirMeansDown = true): Operator {
     const dir = Dimension.direction()
-    const dimX = this.dimX
-    const dimY = this.dimY
     const s = yDirMeansDown ? 1 : -1
 
     return Operator.add([
-      Operator.outer([Operator.shift(dimX, +1), Operator.identity([dimY]), Operator.indicator([dir], ['>'])]),
-      Operator.outer([Operator.shift(dimX, -1), Operator.identity([dimY]), Operator.indicator([dir], ['<'])]),
-      Operator.outer([Operator.identity([dimX]), Operator.shift(dimY, +s), Operator.indicator([dir], ['v'])]),
-      Operator.outer([Operator.identity([dimX]), Operator.shift(dimY, -s), Operator.indicator([dir], ['^'])]),
+      Operator.outer([Operator.shift(this.dimX, +1), Operator.identity([this.dimY]), Operator.indicator([dir], ['>'])]),
+      Operator.outer([Operator.shift(this.dimX, -1), Operator.identity([this.dimY]), Operator.indicator([dir], ['<'])]),
+      Operator.outer([Operator.identity([this.dimX]), Operator.shift(this.dimY, +s), Operator.indicator([dir], ['v'])]),
+      Operator.outer([Operator.identity([this.dimX]), Operator.shift(this.dimY, -s), Operator.indicator([dir], ['^'])]),
```

Comment:
Well, I did that change the other way.

Reply:
It's considered a bad pattern but you seem to like it :p